src/code/z_vimode.c

```diff
@@ -1,17 +1,214 @@
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_vimode/func_800AC030.s")
+void ViMode_LogPrint(OSViMode* viMode) {
+    OSViMode** viMode_ptr = &viMode;
+#define _LOG(line, fmt_str, val)                 \
+    LogUtils_LogThreadId("../z_vimode.c", line); \
+    osSyncPrintf(fmt_str, val)
+    _LOG(87, "osvimodep = %08x\n", (*viMode_ptr));
+    _LOG(88, "osvimodep->comRegs.ctrl = %08x\n", (*viMode_ptr)->comRegs.ctrl);
+    _LOG(89, "osvimodep->comRegs.width = %08x\n", (*viMode_ptr)->comRegs.width);
+    _LOG(90, "osvimodep->comRegs.burst = %08x\n", (*viMode_ptr)->comRegs.burst);
+    _LOG(91, "osvimodep->comRegs.vSync = %08x\n", (*viMode_ptr)->comRegs.vSync);
+    _LOG(92, "osvimodep->comRegs.hSync = %08x\n", (*viMode_ptr)->comRegs.hSync);
+    _LOG(93, "osvimodep->comRegs.leap = %08x\n", (*viMode_ptr)->comRegs.leap);
+    _LOG(94, "osvimodep->comRegs.hStart = %08x\n", (*viMode_ptr)->comRegs.hStart);
+    _LOG(95, "osvimodep->comRegs.xScale = %08x\n", (*viMode_ptr)->comRegs.xScale);
+    _LOG(96, "osvimodep->fldRegs[0].vStart = %08x\n", (*viMode_ptr)->fldRegs[0].vStart);
+    _LOG(97, "osvimodep->fldRegs[0].vBurst = %08x\n", (*viMode_ptr)->fldRegs[0].vBurst);
+    _LOG(98, "osvimodep->fldRegs[0].origin = %08x\n", (*viMode_ptr)->fldRegs[0].origin);
+    _LOG(99, "osvimodep->fldRegs[0].yScale = %08x\n", (*viMode_ptr)->fldRegs[0].yScale);
+    _LOG(100, "osvimodep->fldRegs[0].vIntr = %08x\n", (*viMode_ptr)->fldRegs[0].vIntr);
+    _LOG(101, "osvimodep->fldRegs[1].vStart = %08x\n", (*viMode_ptr)->fldRegs[1].vStart);
+    _LOG(102, "osvimodep->fldRegs[1].vBurst = %08x\n", (*viMode_ptr)->fldRegs[1].vBurst);
+    _LOG(103, "osvimodep->fldRegs[1].origin = %08x\n", (*viMode_ptr)->fldRegs[1].origin);
+    _LOG(104, "osvimodep->fldRegs[1].yScale = %08x\n", (*viMode_ptr)->fldRegs[1].yScale);
+    _LOG(105, "osvimodep->fldRegs[1].vIntr = %08x\n", (*viMode_ptr)->fldRegs[1].vIntr);
+#undef _LOG
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_vimode/func_800AC2F4.s")
+// This function configures the custom VI mode (`viMode.customViMode`) based
+// on the other flags in `viMode`.
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_vimode/ViMode_Configure.s")
```

Comment:
Curious why this function wasn't attempted. Ideally we like to have all functions at least attempted, and wrap #NON_MATCHING or #NON_EQUIVALENT around the code if it doesn't OK, but it's fine if you'd rather not do it.

Reply:
Ah, I'll wrap it in `NON_MATCHING` - I didn't pick up on that convention. Thanks!

Reply:
Do I have to update the spec file? None of the files I deleted were referenced by the spec file. (The only file I modified that is referenced by the spec file is `z_vimode.o` afaict?)

Reply:
Forget about what I said. That recommendation is only for overlays/actors. My bad :sweat_smile: